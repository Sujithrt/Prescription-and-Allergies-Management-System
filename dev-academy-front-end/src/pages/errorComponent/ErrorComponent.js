import React from 'react';
import Alert from 'react-bootstrap/Alert';
import './../errorComponent/errorComponent.css';
import NavBar from './../../components/navBar/NavBar';

/**
 * Renders an error page when user tries to access and undefined route
 * 
 */
function ErrorComponent() {
	return (
		<>
			<NavBar />
			<Alert variant="warning" className="error">
				An Error Occurred. Please Contact Support at 123-456-789
			</Alert>
		</>
	);
}

export default ErrorComponent;
