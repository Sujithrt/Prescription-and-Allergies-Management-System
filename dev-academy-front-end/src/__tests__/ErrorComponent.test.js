import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow } from 'enzyme';
import ErrorComponent from '../pages/errorComponent/ErrorComponent'

describe("ErrorComponent Test", ()=>{
  it("ErrorComponent is rendered as expected", ()=>{
    const errorComponent = shallow(<ErrorComponent />);
    expect(shallowToJson(errorComponent)).toMatchSnapshot();
  });
});