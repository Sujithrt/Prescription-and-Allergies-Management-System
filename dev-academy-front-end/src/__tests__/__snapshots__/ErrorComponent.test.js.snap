// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent Test ErrorComponent is rendered as expected 1`] = `
<Fragment>
  <PrimarySearchAppBar />
  <Alert
    className="error"
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "render": [Function],
      }
    }
    variant="warning"
  >
    An Error Occurred. Please Contact Support at 123-456-789
  </Alert>
</Fragment>
`;
